presto-iceberg/src/test/java/com/facebook/presto/iceberg/nessie/TestNessieMultiBranching.java

```diff
@@ -0,0 +1,169 @@
+
+import com.facebook.presto.Session;
+import com.facebook.presto.iceberg.IcebergQueryRunner;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.testing.containers.NessieContainer;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.google.common.collect.ImmutableMap;
+import org.projectnessie.client.api.NessieApiV1;
+import org.projectnessie.client.http.HttpClientBuilder;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.LogResponse;
+import org.projectnessie.model.Reference;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.facebook.presto.iceberg.CatalogType.NESSIE;
+import static com.facebook.presto.testing.MaterializedResult.resultBuilder;
+import static com.facebook.presto.tests.QueryAssertions.assertEqualsIgnoreOrder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestNessieMultiBranching
+        extends AbstractTestQueryFramework
+{
+    private NessieContainer nessieContainer;
+    private NessieApiV1 nessieApiV1;
+
+    @BeforeClass
+    public void init()
+            throws Exception
+    {
+        nessieContainer = NessieContainer.builder().build();
+        nessieContainer.start();
+        nessieApiV1 = HttpClientBuilder.builder().withUri(nessieContainer.getRestApiUri()).build(NessieApiV1.class);
+        super.init();
+    }
+
+    @AfterClass
+    public void tearDown()
+    {
+        if (null != nessieContainer) {
+            nessieContainer.stop();
+        }
+
+        if (null != nessieApiV1) {
+            nessieApiV1.close();
+        }
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return IcebergQueryRunner.createIcebergQueryRunner(ImmutableMap.of(), ImmutableMap.of("iceberg.catalog.type", NESSIE.name(), "iceberg.nessie.uri", nessieContainer.getRestApiUri()));
+    }
+
+    @Test
+    public void testWithUnknownBranch()
+    {
+        assertQueryFails(sessionOnRef("unknownRef"), "CREATE SCHEMA nessie_namespace", ".*Nessie ref 'unknownRef' does not exist. This ref must exist before creating a NessieCatalog.");
+    }
+
+    @Test
+    public void testNamespaceVisibility()
+            throws NessieConflictException, NessieNotFoundException
+    {
+        Reference one = createBranch("branchOne");
+        Reference two = createBranch("branchTwo");
+        Session sessionOne = sessionOnRef(one.getName());
+        Session sessionTwo = sessionOnRef(two.getName());
+        assertQuerySucceeds(sessionOne, "CREATE SCHEMA namespace_one");
+        assertQuerySucceeds(sessionOne, "SHOW SCHEMAS FROM iceberg LIKE 'namespace_one'");
```

Comment:
this and the one below will be changed to `assertThat(computeActual(sessionOne, "SHOW SCHEMAS FROM iceberg LIKE 'namespace_one'").getMaterializedRows()).hasSize(1);` once Iceberg 0.14.0 is out, since it requires https://github.com/apache/iceberg/pull/4700 to properly work